python/ray/rllib/ddpg/models.py

```diff
@@ -0,0 +1,185 @@
+                registry, self.obs_tp1, state_space, ac_space, self.a_tp1, config)
+            self.ct_var_list = _scope_vars(scope.name)
+
+        y_i = self.rew_t + config["gamma"] * self.q_tp1
+
+        # compute the  error
+
+        self.td_error = tf.losses.mean_squared_error(labels=y_i, predictions=self.q_t)
+        self.action_lost = - tf.reduce_mean(self.q_t)
+
+        self.loss_inputs = [
+            ("obs", self.obs_t),
+            ("rewards", self.rew_t),
+            ("new_obs", self.obs_tp1),
+            ("dones", self.done_mask),
+            ("weights", self.importance_weights),
+        ]
+        self.a_grads = tf.gradients(self.action_lost, self.a_var_list)
+        self.a_grads_and_vars = list(zip(self.a_grads, self.a_var_list))
+        # self.c_grads = tf.gradients(self.td_error, self.c_var_list)
+        # self.c_grads_and_vars = list(zip(self.c_grads, self.c_var_list))
+
+        self.c_grads = optimizer.minimize(self.td_error, var_list=self.c_var_list)
+
+        self.train_expr = optimizer.apply_gradients(self.a_grads_and_vars)
+
+        update_target_expr = []
+        for ta, ea, tc, ec in zip(self.at_var_list, self.a_var_list, self.ct_var_list, self.c_var_list):
+            update_target_expr.append(ta.assign(config["tau"] * ea + (1-config["tau"]) * ta))
+            update_target_expr.append(tc.assign(config["tau"] * ec + (1 - config["tau"]) * tc))
+        self.update_target_expr = tf.group(*update_target_expr)
+
+    def update_target(self, sess):
+
+        return sess.run(self.update_target_expr)
+
+    def copy_target(self, sess):
+        copy_target_expr = []
+        for ta, ea, tc, ec in zip(self.at_var_list, self.a_var_list, self.ct_var_list, self.c_var_list):
+            copy_target_expr.append(ta.assign(ea))
+            copy_target_expr.append(tc.assign(ec))
+        copy_target = tf.group(*copy_target_expr)
+        return sess.run(copy_target)
+
+    def act(self, sess, obs, eps):
+        actor_tf = self.a_t
+        return sess.run(
+            actor_tf,
+            feed_dict={
+                self.obs_t: obs,
+                self.eps: eps,
+            })
+
+    def compute_gradients(
+            self, sess, obs_t, rew_t, obs_tp1, done_mask):
+
+        self.a_grads = [g for g in self.a_grads if g is not None]
+        grads, _ = sess.run(
+            [self.a_grads, self.c_grads],
+
+            feed_dict={
+                self.obs_t: obs_t,
+                self.rew_t: rew_t,
+                self.obs_tp1: obs_tp1,
+                self.done_mask: done_mask,
+            })
+        return grads
+
+    def apply_gradients(self, sess, grads):
+        assert len(grads) == len(self.a_grads_and_vars)
+        feed_dict = dict(zip(self.a_grads, grads))
+
+        sess.run(self.train_expr, feed_dict=feed_dict)
+
+    def _build_q_network(self, registry, inputs, state_space, ac_space, act_t, config):
+        x = inputs
+        x = tf.layers.dense(x, 64)
```

Comment:
can you use tf.slim instead? it would be great to avoid mixing higher level libraries